lib/MooX/Options/Role.pm

```diff
@@ -314,18 +314,15 @@ sub parse_options {
         }
     }
 
-    if (   $opt->help() || defined $params{help}
```

Comment:
the attributes is defined in the role, but I don't see patch that remove them when you don't want them. Also you may have forget the check on existence of those methods if option is disabled.